public/app/plugins/panel/nodeGraph/editor/RawFieldNameSelect.tsx

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+
+import { StandardEditorProps } from '@grafana/data';
+import { Select } from '@grafana/ui';
+
+import { NodeGraphOptions } from '../types';
+
+type Settings = { placeholder?: string };
+type RawFieldNameSelectProps = StandardEditorProps<string, Settings, NodeGraphOptions, undefined>;
+
+export const RawFieldSelector = ({ value, onChange, context, item }: RawFieldNameSelectProps) => {
+  const fieldOptions = [];
+  for (const frame of context.data) {
```

Comment:
Not exactly sure how this works but it seems like I get all the fields from both node and edge data frame in the options:
<img width="368" alt="Screenshot 2023-12-27 at 18 48 30" src="https://github.com/grafana/grafana/assets/1014802/0ca89983-7b04-44d2-a979-7618a333c0df">

Reply:
Hmm, yes you are right. When the names collide choosing the correct field from the correct dataframe could be problematic.

Reply:
Hmm, this gets even more complicated. Because if the edge dataframe isn't named edges and there is no field named source, the dataframe cannot be identified as the dataframe containing edges. If we want this to work for any data, you would have to be able to specify the frame to be used as edges, and the frame to be used as nodes.

Reply:
I've updated the PR so that it's possible to select the frame for nodes and edges. It should also only display fieldnames from the correct frame. However, it feels like this is getting very complicated now :/